rust/lsp-lib/src/lsp_plugin.rs

```diff
@@ -372,6 +365,102 @@ impl Plugin for LspPlugin {
     }
 
     fn config_changed(&mut self, _view: &mut View<Self::Cache>, _changes: &ConfigTable) {}
+
+    fn get_hover(
+        &mut self,
+        view: &mut View<Self::Cache>,
+        request_id: usize,
+        position: CorePosition,
+    ) {
+        let rev = view.rev;
+        let view_info = self.view_info.get_mut(&view.get_id());
+        let view_id = view.get_id();
+
+        let view_info = if let Some(view_info) = view_info {
+            view_info
+        } else {
+            return;
+        };
+
+        // This won't fail since we definitely have a client for the given
+        // client identifier
+        let ls_client = self
+            .language_server_clients
+            .get(&view_info.ls_identifier)
+            .unwrap();
+
+        let position_ls = lsp_position_from_core_position(view, position);
+        let mut ls_client = ls_client.lock().unwrap();
+
+        match position_ls {
+            Ok(position) => {
+                ls_client.request_hover(
+                    view_id,
+                    position,
+                    move |ls_client, result| {
+                        let res = result
+                            .map_err(|e| LanguageResponseError::LanguageServerError(format!("{:?}", e)))
+                            .and_then(|h| {
+                                let hover: Option<Hover> = serde_json::from_value(h).unwrap();
+                                hover.ok_or(LanguageResponseError::NullResponse)
+                                    .and_then(core_hover_from_hover)
+                            });
+                        ls_client.core.display_hover(view_id, request_id, res, rev);
+                    },
+                );
+            }
+            Err(error) => {
+                eprintln!("Can't convert location to offset. Error {:?}", error);
+                let err = LanguageResponseError::PositionConversionError(format!("{:?}",error));
```

Comment:
I would prefer to impl `From<_>` to do this conversion, and then we could just have `ls_client.core.display_hover(..., err.into())`; although I have bigger questions about this, see my top level comment.